### PR TITLE
[WIP] [NOT READY] Offline support

### PR DESCRIPTION
**Pull request by patrick91**
Hello, I did some work to see if it was easy to add support for offline to this plugin. It seems possible, but I'm not sure how we could fetch the lambda functions referred in the datasources, we are using `Fn::GetAtt` so we don't have a direct reference to the function in the Serverless configuration. @sid88in do you have any suggestion?

# TODO

- [ ] allow configuration (port, host)
- [ ] is offline ok as command? it conflicts with serverless offline
- [ ] fetch resolvers

### CODE REVIEW

**Comment by sid88in** [comment]
@patrick91 this is great! @pmuens any help help with @patrick91's question? I am not entire sure.

**Comment by patrick91** [comment]
@sid88in could be a good idea (even if it will break backwards compat) to change the way we reference the lambda functions if they are defined in `serverless.yml`?

File: index.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
 const fs = require('fs');
+const Hapi = require('hapi');
```
Comment (lightsofapollo) [inline index.js]: Curious why hapi and not express/plain http/etc. ?

File: index.js [hunk 2]
```diff
@@ -23,6 +26,11 @@ class ServerlessAppsyncPlugin {
         usage: 'DEPRECATED: Helps you update AppSync API',
         lifecycleEvents: ['update'],
       },
+      // TODO: this conflicts with serverless offline
```
Comment (lightsofapollo) [inline index.js]: Curious what @sid88in thinks but my gut feeling is to name it `appsync-offline`.

**Comment by lightsofapollo** [comment]
@patrick91 Code generally LGTM but I am curious what your ultimate intent is. My team and I also wanted to use appsync offline so we wrote a full featured [emulator](https://github.com/ConduitVC/aws-utils/tree/master/packages/appsync-emulator-serverless) that at least today relies on the serverless.yml config used by this plugin. Are you headed in that direction or where you looking to build some minimal functionality ?

File: index.js (hunk 1 above)
Comment (patrick91) [reply index.js]: I've got inspiration from serverless offline, I'm not super familiar with the node ecosystem :)

**Comment by patrick91** [comment]
@lightsofapollo I didn't know about your emulator! I wanted to have something like serverless-offline for serverless-appsync so that my coworker could run the backend locally without too much hassle

File: index.js (hunk 2 above)
Comment (sid88in) [reply index.js]: 👍 sounds good!

**Comment by sid88in** [comment]
@lightsofapollo @patrick91 maybe you guys can join forces to include this feature in this plugin? Will be simple to integrate just one plugin with more features and others can enhance it too? What do you guys think?

**Comment by lightsofapollo** [comment]
@sid88in I would still consider the emulator to be "alpha" quality. I would like to have my team bang on it (would welcome help!!!) for another week. If you're open to it I would like to include the emulator as "offline" mode. 

As a footnote: Including the emulator would increase the size of this package significantly (just the dynamodb emulator downloads around 17mb).

@patrick91 Happy to collaborate. If you add incremental functionality here that is useful I don't want to hold you up waiting for our stuff. Maybe we could sync briefly about what your `serverless.yml` looks like and see if the emulator in it's current state could be useful to you.

**Comment by sid88in** [comment]
@patrick91 any updates on this?

**Pull request closed without merging**